services/avtransport/avtransport.go

```diff
@@ -14,7 +14,7 @@ import (
 )
 
 type Client struct {
-	HTTPClient *http.Client
```

Comment:
This is a public API change. The HTTPClient is exported so library consumers can replace it with a RetryClient, etc. And I do so in Supersonic (because some UPnP devices are surprisingly flaky)

Reply:
I need to be able to use the HashiCorp RetryClient - see https://github.com/dweymouth/supersonic/blob/main/backend/player/dlna/dlnaplayer.go#L92

One pattern we've used at my work is the functional option pattern in the constructor with an option to accept a different HTTPClient (via an interface that declares only the needed methods). What do you think about that?

Reply:
I have coded up an alternative solution now :)

Reply:
I think defining the RequestHandler type as an interface with the one function, rather than a function pointer directly, could be better, as that allows you to directly assign `RequestHandler: &http.Client{... }` rather than `RequestHandler: (&http.Client{... }).Do` What do you think?